kratos/utilities/sub_model_part_operations_utility.h

```diff
@@ -0,0 +1,165 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Miguel Maso Sotomayor
+//
+
+#ifndef KRATOS_SUB_MODEL_PART_OPERATIONS_H_INCLUDED
+#define KRATOS_SUB_MODEL_PART_OPERATIONS_H_INCLUDED
+
+
+// External includes
+
+
+// Project includes
+#include "includes/model_part.h"
+#include "processes/entity_erase_process.h"
+
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class SubModelPartOperationsUtility
+ * @brief Wrapper of set operations: union, intersection and difference
+ * @author Miguel Maso Sotomayor
+ * @ingroup KratosCore
+*/
+class KRATOS_API(KRATOS_CORE) SubModelPartOperationsUtility
```

Comment:
```suggestion
class KRATOS_API(KRATOS_CORE) ModelPartOperationsUtility
```

Nothing of your code is related to submodelparts

Reply:
in principle, it are sub_model_parts and belong to the same root model part.
The destination model part must be sub model part, and the first and second should be sub model parts

Reply:
https://github.com/KratosMultiphysics/Kratos/blob/28ae221ad973564f6a0dbf82774a814910f159db/kratos/utilities/sub_model_part_operations_utility.h#L152

Reply:
Hm then I would say you need to add some check to ensure this

I was already thinking of using this for different MainModelParts :)

Reply:
The check is made at python level, and having everything in the same root model part make things easier

Reply:
In the first design I started working with different model parts and implementing the operations with a local flag. But then I moved to the std implementation and sub model parts

Reply:
Here the entities are compared by their Id, if they are different MainModelParts, the comparison would be HasIntersection

Reply:
I've added a check at c++ level. I've allowed the first and second model parts to be root. Maybe someone wants it to compute the complementary:
```json
        "Parameters"            : {
            "first_model_part_name"  : "Main",
            "second_model_part_name" : "Main.set",
            "result_model_part_name" : "Main.complementary",
            "set_operation"          : "Difference",
            "entity_type"            : "Elements"
        }
```